qutip/lattice.py

```diff
@@ -0,0 +1,1286 @@
+            r_shape = (self._length_of_unit_cell, self._length_of_unit_cell)
+            if cell_Hamiltonian.shape != r_shape:
+                raise Exception("cell_Hamiltonian does not have a shape \
+                            consistent with cell_num_site and cell_site_dof.")
+            self._H_intra = Qobj(cell_Hamiltonian, dims=dim_ih)
+
+        is_real = np.isreal(self._H_intra).all()
+
+        nSb = self._H_intra.shape
+        if isinstance(inter_hop, list):      # There is a user input list
+            for i in range(len(inter_hop)):
+                if not isinstance(inter_hop[i], Qobj):
+                    raise Exception("\ninter_hop[", i, "] is not a Qobj. All \
+                                inter_hop list elements need to be Qobj's. \n")
+                nSi = inter_hop[i].shape
+                # inter_hop[i] is a Qobj, now confirmed
+                if nSb != nSi:
+                    raise Exception("\ninter_hop[", i, "] is dimensionally \
+                        incorrect. All inter_hop list elements need to \
+                        have the same dimensionality as cell_Hamiltonian.")
+                else:    # inter_hop[i] has the right shape, now confirmed,
+                    inter_hop[i] = Qobj(inter_hop[i], dims=dim_ih)
+                    is_real = is_real and np.isreal(inter_hop[i]).all()
+            self._H_inter_list = inter_hop    # The user input list was correct
+            # we store it in _H_inter_list
+        elif isinstance(inter_hop, Qobj):  # There is a user input
+            # Qobj
+            nSi = inter_hop.shape
+            if nSb != nSi:
+                raise Exception("inter_hop is required to have the same \
+                dimensionality as cell_Hamiltonian.")
+            else:
+                inter_hop = Qobj(inter_hop, dims=dim_ih)
+            self._H_inter_list = [inter_hop]
+            is_real = is_real and np.isreal(inter_hop).all()
+
+        elif inter_hop is None:      # inter_hop is the default None)
+            # So, we set self._H_inter_list from cell_num_site and
+            # cell_site_dof
+            if cell_num_site == 1:
+                siteT = Qobj([[-1]])
+            else:
+                bNm = basis(cell_num_site, cell_num_site-1)
+                bN0 = basis(cell_num_site, 0)
+                siteT = bNm * bN0.dag() + bN0 * bNm.dag()
+            inter_hop = tensor(Qobj(siteT), qeye(self.cell_site_dof))
+            self._H_inter_list = [inter_hop]
+        else:
+            raise Exception("\n\n inter_hop is required to be a Qobj or a \
+                            list of Qobjs.")
+
+        self.positions_of_sites = [(i/self.cell_num_site) for i in
+                                   range(self.cell_num_site)]
+        self._is_consistent = self._checks(check_if_complete=True)
+        self._inter_vec_list = [[1] for i in range(len(self._H_inter_list))]
+        self._Brav_lattice_vectors_list = [[1]]     # unit vectors
```

Comment:
I think, this will be useful for clarity of the method. Although never used, it is meaningful to keep this information in the lattice class, to make the lattice class a complete description of a crytal.